stonesoup/simulator/transition.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+from datetime import timedelta
+from copy import deepcopy
+
+import numpy as np
+
+from stonesoup.models.transition.linear import ConstantTurn, ConstantVelocity,\
+    CombinedLinearGaussianTransitionModel
+from stonesoup.types.array import StateVector
+
+
+def create_smooth_transition_models(initial_state, x_coords, y_coords, times, turn_rate):
+    """Generate a list of constant-turn and constant acceleration transition models alongside a
+    list of transition times to provide smooth transitions between 2D cartesian coordinates and
+    time pairs.
+    An assumption is that the initial_state's x, y coordinates are the first elements of x_ccords
+    and y_coords respectively. Ie. The platform starts at the first coordinates.
+
+    Parameters
+    ----------
+    initial_state: :class:`~.State` The initial state of the platform.
+    x_coords:
+        A list of int/float x-coordinates (cartesian) in the order that the target must follow.
+    y_coords:
+        A list of int/float y-coordinates (cartesian) in the order that the target must follow.
+    times:
+        A list of :class:`~.datetime.datetime` dictating the times at which the target must be at
+        each corresponding coordinate.
+    turn_rate: Float
+        Angular turn rate (radians/second) measured anti-clockwise from positive x-axis.
```

Comment:
Is this a maximum, preferred or fixed turn rate (i.e. will this always be the same rate even if a small turn is needed)?

Reply:
This is a fixed turn rate. Variable turn rate will require a bit more work but is something I'm interested in adding. Perhaps providing a max and min turn rate and having some logic to decide what to choose inbetween?

Reply:
So the platform is always either turning **or** accelerating? Never both?

Reply:
Yes, quite a basic set of transition models, but allowing speed change during turn seems complicated.